tests/python/test_dynamic.py

```diff
@@ -283,3 +264,51 @@ def make_list():
             assert x[i, j].b == i * j * 10000 % 65536
             assert x[i, j].c == i * j * 100000000 % 4294967296
             assert x[i, j].d == i * j * 10000000000
+
+
+@test_utils.test(require=ti.extension.sparse, exclude=[ti.metal])
```

Comment:
Can you try to add another test with `real_matrix=True, real_matrix_scalarize=True`? Something like:
```
@test_utils.test(require=ti.extension.sparse, exclude=[ti.metal], real_matrix=True, real_matrix_scalarize=True)
def test_append_matrix_scalarize():
     _test_append_matrix()
```

I can help with the fix if it fails.